root/static/scripts/account/components/ApplicationForm.js

```diff
@@ -57,18 +57,19 @@ class ApplicationForm extends React.Component<Props, State> {
 
   handleOauthRedirectURIChange(e: SyntheticEvent<HTMLInputElement>) {
     const selectedOauthRedirectURI = e.currentTarget.value;
-    this.setState(prevState => mutate<State, _>(prevState, newState => {
-      newState.form.field.oauth_redirect_uri.value = selectedOauthRedirectURI;
-    }));
+    this.setState(prevState => mutate(prevState)
+      .set('form', 'field', 'oauth_redirect_uri', 'value',
+           selectedOauthRedirectURI)
+      .final());
   }
 
   handleOauthTypeChangeBound: (e: SyntheticEvent<HTMLSelectElement>) => void;
 
   handleOauthTypeChange(e: SyntheticEvent<HTMLSelectElement>) {
     const selectedOauthType = e.currentTarget.value;
-    this.setState(prevState => mutate<State, _>(prevState, newState => {
-      newState.form.field.oauth_type.value = selectedOauthType;
-    }));
+    this.setState(prevState => mutate(prevState)
+      .set('form', 'field', 'oauth_type', 'value', selectedOauthType)
+      .final());
```

Comment:
It returns the final copy of the object with all the changes made (and if it object is frozen it re-freezes it). It's final 'cause it ends the mutation and is the final thing you call. :) Nothing stops you from passing the result to `mutate` again (since the point is that it can modify read-only objects).